File: BINARYIO - Binary Input and Output/Binary Input and Output_v2.cpp
/*
* Project name : SPOJ: BINARYIO - Binary Input and Output
* Author       : Wojciech Raszka
* Date created : 2019-03-10
* Description  :
* Status       : Accepted (23379855)
* Comment      :
*/

#include <stdio.h>
#include <math.h>

int main(){
  const int MAX_SIZE = 1000000;
  double x[MAX_SIZE];

  size_t no_of_ret = fread(x, sizeof *x, MAX_SIZE, stdin);
  for (int i = 0; i < no_of_ret; i++){
      x[i] = log(x[i]);
  }

  fwrite(x, sizeof *x, no_of_ret, stdout);

  return 0;
}

File: BINARYIO - Binary Input and Output/Binary Input and Output_v2.py
# Project name : SPOJ: BINARYIO - Binary Input and Output
# Author       : Wojciech Raszka
# Date created : 2019-03-10
# Description  :
# Status       : Accepted (23380003)
# Comment      :

import sys
import struct
from math import log

inp = sys.stdin.buffer.read()
no_of_ret = len(inp)//8
sys.stdout.buffer.write(struct.pack("%sd" % no_of_ret, *(list(map(lambda x: log(x), struct.unpack("%sd" % no_of_ret, inp))))))
